cnxarchive/database.py

```diff
@@ -73,12 +74,18 @@ def _read_sql_file(name):
     }
 
 
+@contextlib.contextmanager
 def db_connect(connection_string=None):
     """Function to supply a database connection object."""
     if connection_string is None:
         settings = get_current_registry().settings
         connection_string = settings[config.CONNECTION_STRING]
-    return psycopg2.connect(connection_string)
+    db_conn = psycopg2.connect(connection_string)
+    try:
+        with db_conn:
+            yield db_conn
+    finally:
+        db_conn.close()
```

Comment:
This is probably good for now, but it occurs to me that it provides for the possibility of doing some connection pooling here.